tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/AWS/SNS/CachedMessageHeadersHelper.cs

```diff
@@ -23,8 +25,8 @@ internal static class CachedMessageHeadersHelper<TMarkerType>
         static CachedMessageHeadersHelper()
         {
             // Initialize delegate for creating a MessageAttributeValue object
-            var messageAttributeValueType = typeof(TMarkerType).Assembly.GetType("Amazon.SimpleNotificationService.Model.MessageAttributeValue");
-            var messageAttributeValueCtor = messageAttributeValueType.GetConstructor(System.Type.EmptyTypes);
+            var messageAttributeValueType = typeof(TMarkerType).Assembly.GetType("Amazon.SimpleNotificationService.Model.MessageAttributeValue")!;
```

Comment:
In this file I am asserting that even though the Reflection API's return nullable reference types (e.g. `System.Type?`) we expect them to be non-null (e.g. `System.Type`). Just want to call that out here